tests/sources/fixtures/github/fixture.py

```diff
@@ -0,0 +1,221 @@
+        self.app.route(
+            "/repos/demo_user/demo_repo/git/blobs/<string:commit_id>", methods=["GET"]
+        )(self.get_content)
+        self.app.route("/repos/demo_user/demo_repo/commits", methods=["GET"])(
+            self.get_commits
+        )
+
+    def get_user(self):
+        response = make_response(
+            {
+                "login": "demo_user",
+                "id": "user_1",
+            }
+        )
+        response.status_code = 200
+        response.headers["X-OAuth-Scopes"] = ["repo"]
+        return response
+
+    def get_repos(self):
+        args = request.args
+        if args.get("type") == ALL_TYPE and args.get("page") == FIRST_PAGE:
+            return [
+                {
+                    "id": "repo_1",
+                    "updated_at": "2023-04-19T09:32:54Z",
+                    "name": "demo_repo",
+                    "full_name": "demo_user/demo_repo",
+                    "html_url": "https://github.com/demo_user/demo_repo",
+                    "owner": {"login": "demo_user"},
+                    "description": "this is a test repository",
+                    "visibility": "public",
+                    "language": "Python",
+                    "fork": "false",
+                    "open_issues": 4,
+                    "default_branch": "main",
+                    "forks_count": 2,
+                    "watchers": 10,
+                    "stargazers_count": 10,
+                    "created_at": "2023-04-19T09:32:54Z",
+                }
+            ]
+        if args.get("page") == "2":
+            return []
+
+    def get_pulls(self):
+        global PULL_REQUEST_PAGES
+        args = request.args
+        if args.get("state") == ALL_TYPE:
+            pulls = [
+                {
+                    "id": f"pulls_{args['page']}_{pull_number}",
+                    "updated_at": "2023-04-24T08:44:00Z",
+                    "number": pull_number,
+                    "html_url": f"https://github.com/demo_user/demo_repo/pull/{pull_number}",
+                    "created_at": "2023-04-24T08:44:00Z",
+                    "closed_at": "2023-04-25T08:44:00Z",
+                    "title": "demo txt",
+                    "state": "open",
+                    "merged_at": "2023-04-25T08:44:00Z",
+                    "merge_commit_sha": "2123bad7d8sfsa4fds54",
+                    "body": "Update hello world",
+                    "user": {"login": "demo_user"},
+                    "head": {"label": "test-branch"},
+                    "base": {"label": "main"},
+                    "assignees": [{"login": "demo_user"}],
+                    "requested_reviewers": [{"login": "test_user"}],
+                    "requested_teams": [{"name": "team1"}],
+                    "labels": [{"name": "V8.8"}],
+                }
+                for pull_number in range(100)
+            ]
+        if args.get("page") == PULL_REQUEST_PAGES:
+            PULL_REQUEST_PAGES = DELETED_PAGES
```

Comment:
In python all caps variables are considered constants by convention. So I would probably use a field here as constants shouldn't be changed.